frappe/core/doctype/user/user.js

```diff
@@ -73,6 +73,14 @@ frappe.ui.form.on('User', {
 
 		if(!frm.is_new()) {
 			if(has_access_to_edit_user()) {
+				frm.add_custom_button(__("Resend Welcome Email"), function(){
```

Comment:
Why not use frm.call to directly call resend_welcome_email rather than writing a new whitelisted method. Refer this